manifests/config.pp

```diff
@@ -44,7 +44,15 @@
 
   case $ensure {
     'present': {
-      $changes = "set ${name} '${value}'"
+      if $comment {
+        $changes = [
+          "set ${name} '${value}'",
+          "insert #comment before $name",
```

Comment:
Hmm...I'll reset this soon as I get a chance.  If you would like I can cancel this PR until further testing.